### PR TITLE
Allow GitLab connections without hook permissions

### PR DESCRIPTION
**Pull request by AndrewFerr**
Warn instead of fail when connecting a GitLab project that Hookshot cannot provision a webhook for.

---

This also updates documentation to explain that "Maintainer" or higher permissions are required on a GitLab project for Hookshot to be able to add a webhook to it.

The warning appears for both `hookshot gitlab project <url>` and the widget:
Screenshot from 2022-11-04 11-38-31

### CODE REVIEW

**Comment by AndrewFerr** [comment]
Fixes #545

File: web/components/roomConfig/RoomConfig.tsx [hunk 1]
```diff
@@ -86,7 +93,7 @@ export const RoomConfig = function<SConfig, ConnectionType extends GetConnection
 
     return <main>
         {
-            error && <ErrorPane header={error.header || "Error"}>{error.message}</ErrorPane>
+            error && <ErrorPane header={error.header || "Error"} isWarning={error.isWarning}>{error.message}</ErrorPane>
```
Comment (Half-Shot) [inline web/components/roomConfig/RoomConfig.tsx:96]: I'd either do <ErrorPane kind="warning"> *or* <WarningPane> (preference on the latter). `isWarning` feels clunky.

File: web/components/elements/ErrorPane.tsx [hunk 2]
```diff
@@ -1,9 +1,10 @@
...
-import ErrorBadge from "../../icons/warning-badge.svg";
+import ErrorBadge from "../../icons/error-badge.svg";
+import WarningBadge from "../../icons/warning-badge.svg";
 import style from "./ErrorPane.module.scss";
 
-export const ErrorPane: FunctionComponent<{header?: string}> = ({ children, header }) => {
-    return <div class={`card error ${style.errorPane}`}>
-        <p><strong><img src={ErrorBadge} /> { header || "Error occured during widget load" }</strong>: {children}</p>
+export const ErrorPane: FunctionComponent<{header?: string, isWarning?: boolean}> = ({ children, header, isWarning }) => {
+    return <div class={`card error ${isWarning ? style.warningPane : style.errorPane}`}>
```
Comment (Half-Shot) [inline web/components/elements/ErrorPane.tsx:7]: Yeah, the `isWarnings here make me feel this should be a separate function :D

File: web/components/elements/ErrorPane.module.scss [hunk 3]
```diff
@@ -1,4 +1,8 @@
 .errorPane {
     max-width: 480px;
     color: #FF4B55;
+}
+.warningPane {
+    max-width: 480px;
+    color: #CCAA00;
```
Comment (Half-Shot) [inline web/components/elements/ErrorPane.module.scss:7]: This colour isn't great for contrast, have you consulted the Figma Oracles?

https://www.figma.com/file/KhVjYAzy4oX3eIqwu9KQxN/Element-library?node-id=4779%3A18811
https://www.figma.com/file/X4XTH9iS2KGJ2wFKDqkyed/Compound?node-id=557%3A0

**Review by Half-Shot — CHANGES_REQUESTED** [review]

**Review by Half-Shot — APPROVED** [review]
Smashing!

**Pull request merged**